### PR TITLE
Fix Cast exception when the "results" stage is disabled in a pipeline.

### PR DESCRIPTION
**Pull request by JulienDevillers**
# Description
Changed a message to something more accurate.

# Justification
Bug

# Instructions for Use
NAF

# Implementation Details
1. How did you test the change? 
used the test case on #826 

2. Did you follow the [coding style](https://github.com/openpnp/openpnp/wiki/Developers-Guide#coding-style)?
Yes

3. If you made changes in the `org.openpnp.spi` or `org.openpnp.model` packages you will need to add additional justification for these changes. Changes to these packages require extensive review and testing.
No

4. Be sure to run `mvn test` before submitting the Pull Request. If the tests do not pass the Pull Request will not be accepted.
Not worse than before.... :D

### CODE REVIEW

**Comment by JulienDevillers** [comment]
Sorry, I should have done the two changes in two pull requests...
First time I do this.

**Comment by markmaker** [comment]
Hi Julien

There might be a misunderstanding. An OpenPNP Computer Vision pipeline has actually two outcomings: 

1. The last enabled stage creates the image to show to the user. 
2. The stage named "results" creates the operational vision result (e.g. a list of points, circles or rectangles etc. i.e. positional information).

They are usually not the same stages!

The "results" stage is where OpenPNP gets its operational results from. You can't disable it, it doesn't make sense. 

If you don't want this  stage to be used, you must rename it into something else and then assign the "results" name to another **enabled** stage that you actually want to get your operational results from. 

If this is not a misunderstanding, please explain further.

_Mark

**Comment by JulienDevillers** [comment]
Hi Mark,There's no misunderstood.Here's the history:I had the bug referenced #826I did not find in the documentation that the stage 'results' was mandatory.Has I did not get answer to my bug request and suspected it should not be to complicated, I downloaded the code and start debugging.While debugging, I understood what you explained me.But the fact is that in the GUI, you can disable the results stage and it leads to a non understoodable message 'cast exception'.You can also remove the results stage, then you get a nullpointer exception.I could have fix it by disabling the ability to uncheck results, but I should have to disable the ability to rename it too. But this solution is not user friendly to me because When you test a pipe, you want something versatile.So I decided, to get rid of the ugly exception message ( it should have been a true exception with call stack ) , and build a user friendly message 'results stage not enabked'.I also fix the null exception by a message 'there should be a results stage in the pipeline'

**Comment by vonnieda** [comment]
Looks good to me, thank you @JulienDevillers!

File: src/main/java/org/openpnp/machine/reference/ReferenceNozzleTip.java [hunk 1]
```diff
@@ -803,19 +803,28 @@ private Location findCircle() throws Exception {
...
-                Object results = pipeline.getResult(VisionUtils.PIPELINE_RESULTS_NAME).model;
+                String stageName = VisionUtils.PIPELINE_RESULTS_NAME;
+                Result pipelineResult = pipeline.getResult(stageName);
+                if (pipelineResult == null) {
+                	throw new Exception(String.format("There should be a \"%s\" stage in the pipeline.", stageName));
+                }
 
+				Object results = pipelineResult.model;
+                
+                if (results instanceof Exception) {
```
[the remark this answers is not in the document]
Comment (JulienDevillers) [reply src/main/java/org/openpnp/machine/reference/ReferenceNozzleTip.java]: No problem. 

Line 812 : Object results = pipelineResult.model;  

line 803 : pipelineResult.model is built by pipeline.process();

CvPipeline.process() looks like this:

 ```
 public void process() {
        totalProcessingTimeNs = 0;
        release();
        for (CvStage stage : stages) {
            // Process and time the stage and get the result.
            long processingTimeNs = System.nanoTime();
            Result result = null;
            try {
                if (!stage.isEnabled()) {
                    throw new Exception(String.format("Stage \"%s\"not enabled.", stage.getName()));
                }
                result = stage.process(this);
            }
            catch (Exception e) {
                result = new Result(null, e);
            }

<snip/>
}

```
If a stage (and in our case, particularly the "results" stage), is not enabled it raises an exception which is catch. "result" is instanciated with the exception as parameter. This parameter is used to set the "model" member of the result. 

So pipelineResult.model and consequently results is, in this particular case, an instance of Exception.

[the remark this answers is not in the document]
Comment (JulienDevillers) [reply src/main/java/org/openpnp/machine/reference/ReferenceNozzleTip.java]: btw, can you explain me what the "resolve conversation" button below does? 
Should I mark the conversation resolved or should you?
I'm pretty new in collaborative github.

**Pull request merged**